lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,268 @@
+                    new_disk = {
+                        'lun': old_disk['lun'],
+                        'disk_size_gb': old_disk['diskSizeGB'],
+                        'managed_disk': old_disk['managedDisk']['id'],
+                        'caching': old_disk['caching'],
+                        'storage_account_type': old_disk['storageAccountType']
+                    }
+                    data_disks[disk_index] = new_disk
+
+                updated = {
+                    'resource_group': self.resource_group,
+                    'name': image['name'],
+                    'location': image['location'],
+                    'tags': image.get('tags'),
+                    'source': image['properties']['sourceVirtualMachine']['id'] if 'sourceVirtualMachine' in image['properties'].keys() else None,
+                    'os_type': image['properties']['storageProfile']['osDisk']['osType'],
+                    'os_state': image['properties']['storageProfile']['osDisk']['osState'],
+                    'os_disk_caching': image['properties']['storageProfile']['osDisk']['caching'],
+                    'os_storage_account_type': image['properties']['storageProfile']['osDisk']['storageAccountType'],
+                    'os_managed_disk': (image['properties']['storageProfile']['osDisk']['managedDisk']['id']
+                                        if 'managedDisk' in image['properties']['storageProfile']['osDisk'].keys() else None),
+                    'provisioning_state': image['properties']['provisioningState'],
+                    'data_disks': data_disks
+                }
+                self.results['ansible_facts']['azure_images'][index] = updated
+
+        return self.results
+
+    def get_image(self, resource_group, image_name):
+        '''
+        Returns image details based on its name
+        '''
+
+        self.log('Get properties for {0}'.format(self.name))
+
+        result = []
+        item = None
+        try:
+            item = self.compute_client.images.get(resource_group, image_name)
+        except CloudError as exc:
+            self.fail('Failed to list images - {0}'.format(str(exc)))
+
+        if item and self.has_tags(item.tags, self.tags):
+            result = [self.serialize_obj(item, 'Image', enum_modules=AZURE_ENUM_MODULES)]
+        return result
+
+    def list_images_by_resource_group(self, resource_group):
+        '''
+        Returns image details based on its resource group
+        '''
+
+        self.log('List images filtered by resource group')
+        results = []
+        response = None
+        try:
+            response = self.compute_client.images.list_by_resource_group(resource_group)
+        except CloudError as exc:
+            self.fail("Failed to list images: {0}".format(str(exc)))
+
+        if response:
+            for item in response:
+                if self.has_tags(item.tags, self.tags):
+                    results.append(self.serialize_obj(item, 'ImagePaged',
```

Comment:
IMO, there is no need to serialize the object if user choose a curated way. So we can only keep the tag filter here as 
```python
result = [x for x in response if self.has_tags(x.tags, self.tags)] if response else []
```